thrall/app/lib/ElasticSearch6.scala

```diff
@@ -0,0 +1,511 @@
+package lib
+
+import com.gu.mediaservice.lib.elasticsearch.ImageFields
+import com.gu.mediaservice.lib.elasticsearch6.{ElasticSearchClient, Mappings}
+import com.gu.mediaservice.lib.formatting.printDateTime
+import com.gu.mediaservice.model._
+import com.gu.mediaservice.model.usage.Usage
+import com.gu.mediaservice.syntax._
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.sksamuel.elastic4s.script.Script
+import com.sksamuel.elastic4s.searches.queries.BoolQuery
+import com.sksamuel.elastic4s.searches.sort.SortOrder
+import org.joda.time.DateTime
+import play.api.libs.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ElasticSearch6(config: ThrallConfig, metrics: ThrallMetrics) extends ElasticSearchVersion with ElasticSearchClient with ImageFields with ElasticSearch6Executions {
+
+  lazy val imagesAlias = config.writeAlias
+  lazy val host = config.elasticsearch6Host
+  lazy val port = 9206
+  lazy val cluster = config("es6.cluster")
+
+  lazy val shards = config.elasticsearch6Shards
+  lazy val replicas = config.elasticsearch6Replicas
+
+  @Deprecated
+  lazy val clientTransportSniff = false
+
+  def indexImage(id: String, image: JsValue)(implicit ex: ExecutionContext): List[Future[ElasticSearchUpdateResponse]] = {
+    // TODO doesn't match the legacy functionality
```

Comment:
How does it not match the old functionality?

Reply:
This implementation is lifted from the lambda branch. It's behaviour appears different to my untrained eye but it could be the same logic with a refactor applied; equally it could be that the lambda branch devs spotted a generalisation which I've missed.

Resolve by porting the Elastic1 implementation of indexImage.